src/djcytoscape/forms.py

```diff
@@ -18,6 +20,23 @@ class CytoscapeGFKChoiceField(AllowedGFKChoiceField):
         what the dynamically loaded list would have produced
     """
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        queryset_models = []
+        for qs in self.queryset.get_querysets():
+            model = qs.model
+            # Do not use Quest, Rank or Badge as an initial object, if it is already an intitial object for another map
+            queryset_models.append(
+                model.objects.exclude(
+                    pk__in=CytoScape.objects.filter(
```

Comment:
Can you provide a test that these objects are excluded? This should be in its own method, fully tested for the correct return values, and from __init__().

Reply:
Calling the `CytoscapeGFKChoiceField()` class constructor creates, and returns a new instance of the class.

First, Python calls `.__new__()` and then `.__init__()`, resulting in a new and fully initialized intance of `CytoscapeGFKChoiceField`.